src/test/groovy/com/google/protobuf/gradle/ProtobufJavaPluginTest.groovy

```diff
@@ -89,8 +92,7 @@ class ProtobufJavaPluginTest extends Specification {
       .withProjectDir(projectDir)
       .withArguments(
           'build', '--stacktrace',
-          '-Dorg.gradle.unsafe.instant-execution=true',
-          '-Dorg.gradle.unsafe.instant-execution.fail-on-problems=true'
+          '--configuration-cache=warn'
```

Comment:
Do we want the test to be stricter by `'--configuration-cache=on'`, although real users can definitely ignore configuration cache problems.

Reply:
I'd love to do that, but the issue is that the OsDetectorPlugin plugin is not fully compliant and it uses JDK APIs to access system properties, instead of the Gradle API exposed with the `ProviderFactory`.

Reply:
Sure, we can live with it now. I was wondering that the OsDetectorPlugin is so small that we can just turn it into a [build service](https://docs.gradle.org/nightly/userguide/build_services.html#build_services) instead of using it as a plugin.

Reply:
That would be great. With that and new Gradle APIs for `tarTree` and `zipTree` in 6.6 we could run with `--configuration-cache=on`.